### PR TITLE
Warn user when keywords version is out of date - for #2174

### PR DESCRIPTION
**Pull request by ismailsunni**
For #2174

### CODE REVIEW

File: safe/gui/widgets/dock.py [hunk 1]
```diff
@@ -1554,10 +1559,45 @@ def show_no_keywords_message(self):
...
+        :type inasafe_version: str
+
+        .. note:: The print button will be disabled if this method is called.
+        """
+        LOGGER.debug('Showing Mismatch Version Message')
+        report = m.Message()
+        report.add(LOGO_ELEMENT)
+        report.add(m.Heading(self.tr(
+            'Layer Keyword\'s Version Mismatch:'), **WARNING_STYLE))
+        context = m.Paragraph(
```
Comment (timlinux) [inline safe/gui/widgets/dock.py]: You layer\'s keyword version (%s) does not match your InaSAFE version (%s). If you wish to use this layer as an exposure, hazard or aggregation layer in an analysis, please use the keyword wizard to update the keywords. You can open the wizard by clicking on the

File: safe/gui/widgets/dock.py [hunk 2]
```diff
@@ -1554,10 +1559,45 @@ def show_no_keywords_message(self):
...
+                ' icon in the toolbar.'))
+        report.add(context)
+        self.pbnPrint.setEnabled(False)
+        # noinspection PyTypeChecker
+        self.show_static_message(report)
+
+    def show_keyword_version_message(
+            self, keyword_version, inasafe_version):
+        """Show a message indicating that the keywords version is mismatch
+
```
Comment (timlinux) [inline safe/gui/widgets/dock.py]: .. versionadded: 3.2

File: safe/utilities/utilities.py [hunk 3]
```diff
@@ -373,3 +374,31 @@ def replace_accentuated_characters(message):
 
     message = unicodedata.normalize('NFKD', message).encode('ASCII', 'ignore')
     return message.decode('utf-8')
+
+
+def compare_version(version1, version2):
+    """Compare between InaSAFE's version.
+
```
Comment (timlinux) [inline safe/utilities/utilities.py]: .. versionadded: 3.2

File: safe/utilities/utilities.py [hunk 4]
```diff
@@ -373,3 +374,31 @@ def replace_accentuated_characters(message):
 
     message = unicodedata.normalize('NFKD', message).encode('ASCII', 'ignore')
     return message.decode('utf-8')
+
+
+def compare_version(version1, version2):
+    """Compare between InaSAFE's version.
```
Comment (timlinux) [inline safe/utilities/utilities.py]: Compare between InaSAFE versions.

**Comment by timlinux** [comment]
LGTM - just please apply the minor docstring and string fixes I provided in the comments.

**Comment by ismailsunni** [comment]
New scrutinizer result : https://scrutinizer-ci.com/g/AIFDR/inasafe/inspections/8b551316-5efd-4040-8f00-7f7bc148ceb8

**Pull request merged**